src/email.c

```diff
@@ -151,7 +151,7 @@ handle_compose_email (XdpImplEmail *object,
   g_variant_lookup (arg_options, "attachments", "^a&s", &attachments);
 
   if (!compose_mail (address, subject, body, attachments, NULL))
-    response = 1;
+    response = 2;
```

Comment:
Wouldn't it be more correct to pass a GError to `compose_mail()` and then return it via D-Bus instead of returning `2`?